CoreLibs/src/org/sleuthkit/autopsy/corelibs/OpenCvLoader.java

```diff
@@ -18,43 +18,31 @@
  */
 package org.sleuthkit.autopsy.corelibs;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
 import org.opencv.core.Core;
 
 public final class OpenCvLoader {
 
+    private static Logger logger = Logger.getLogger(OpenCvLoader.getClass().getName());
     private static final boolean OPEN_CV_LOADED;
-    private static UnsatisfiedLinkError exception = null;
-
+    
     static {
-        boolean tempOpenCvLoaded = false;
         try {
             System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
-            tempOpenCvLoaded = true;
-        } catch (UnsatisfiedLinkError e) {
-            tempOpenCvLoaded = false;
-            exception = e;  //save relevant error for throwing at appropriate time
+            OPEN_CV_LOADED = true;
+        } catch (UnsatisfiedLinkError | SecurityException ex) {
+            OPEN_CV_LOADED = false;
+            logger.log(Level.WARNING, "Unable to load OpenCV", ex);
         }
-        OPEN_CV_LOADED = tempOpenCvLoaded;
     }
 
     /**
      * Return whether or not the OpenCV library has been loaded.
      *
      * @return - true if the opencv library is loaded or false if it is not
-     * @throws UnsatisfiedLinkError - A COPY of the exception that prevented OpenCV from loading.  
-     *   Note that the stack trace in the exception can be confusing because it refers to a
-     *   past invocation. 
      */
-    public static boolean isOpenCvLoaded() throws UnsatisfiedLinkError {
-        if (!OPEN_CV_LOADED) {
-             //exception should never be null if the open cv isn't loaded but just in case
-            if (exception != null) {
-                throw exception;
-            } else {
-                throw new UnsatisfiedLinkError("OpenCV native library failed to load");
-            }
-
-        }
+    public static boolean isOpenCvLoaded() {
```

Comment:
I think this needs to continue to throw an exception. Otherwise it is backward incompatible.

Reply:
From Richard: "I think that in this case we can just modify this exported public API. I can’t imagine anybody is using it but us. It is cool with me if you want to remove the exception specification and the parameter documentation. The proper thing to do would be to deprecate the method and make a new version of it, but it doesn’t seem worth it to me."

Reply:
@dannysmyda, please go ahead and do the deprecation and make a new method so that we preserve this exported public API after all, thanks!